evaluator.plandefinition/src/main/java/org/opencds/cqf/cql/evaluator/plandefinition/r4/PlanDefinitionProcessor.java

```diff
@@ -0,0 +1,725 @@
+    Object result = null;
+    switch (language) {
+      case "text/cql":
+      case "text/cql.expression":
+      case "text/cql-expression": {
+        if (dataRequirements != null && !dataRequirements.isEmpty()) {
+          Parameters params = resolveInputParameters(dataRequirements);
+          if (session.parameters != null && session.parameters instanceof Parameters) {
+            ((Parameters) session.parameters).getParameter().forEach(parameter -> params.addParameter(parameter));
+          }
+          expressionEvaluator.evaluate(expression, params);
+        } else {
+          expressionEvaluator.evaluate(expression, session.parameters);
+        }
+      } break;
+      case "text/cql-identifier":
+      case "text/cql.identifier":
+      case "text/cql.name":
+      case "text/cql-name": {
+        Set<String> expressions = new HashSet<>();
+        expressions.add(expression);
+        result = libraryProcessor.evaluate(libraryToBeEvaluated, session.patientId, session.parameters,
+            session.contentEndpoint, session.terminologyEndpoint, session.dataEndpoint, session.bundle, expressions);
+      } break;
+      case "text/fhirpath": {
+        Parameters params = resolveInputParameters(dataRequirements);
+        if (params != null) {
+          IBaseParameters resultParams = expressionEvaluator.evaluate(expression, params);
+          IBaseDatatype tempResult = operationParametersParser.getValueChild(((Parameters) resultParams), "return");
+          if (tempResult == null) {
+            IBaseResource tempResource = operationParametersParser.getResourceChild(((Parameters) resultParams), "return");
+            result = tempResource;
+          } else {
+            result = tempResult;
+          }
+        } else {
+          result = null;
+        }
+      }
+        break;
+      default:
+        logger.warn("An action language other than CQL was found: " + language);
+    }
+    if (result != null) {
+      if (result instanceof Parameters) {
+        IBaseDatatype tempResult = operationParametersParser.getValueChild(((Parameters) result), expression);
+        if (tempResult == null) {
+          IBaseResource tempResource = operationParametersParser.getResourceChild(((Parameters) result), expression);
+          result = tempResource;
+        } else {
+          result = tempResult;
+        }
+      }
+    }
+    return result;
+  }
+
+  private Parameters resolveInputParameters(List<DataRequirement> dataRequirements) {
+    if (dataRequirements == null || dataRequirements.isEmpty()) {
+      return null;
+    }
+    Parameters params = new Parameters();
+    for (DataRequirement req : dataRequirements) {
+      String name = req.getId();
+      String fhirType = req.getType();
+      Iterator<IBaseResource> searchResult = fhirDal.search(fhirType).iterator();
```

Comment:
We should reconsider this API if we're going to use it for searches such as this. This would load every resource of a given type from the underlying data access layer.